activiti-cloud-query-service/activiti-cloud-services-query/activiti-cloud-services-query-events/src/main/java/org/activiti/cloud/services/query/events/handlers/BaseBPMNActivityEventHandler.java

```diff
@@ -40,38 +42,27 @@ protected BaseBPMNActivityEntity findOrCreateBPMNActivityEntity(CloudRuntimeEven
 
         BaseBPMNActivityEntity bpmnActivityEntity = null;
 
-        if ("serviceTask".equals(bpmnActivity.getActivityType())) {
-            bpmnActivityEntity = entityManager.find(ServiceTaskEntity.class, pkId);
-        } else {
+        if (!SERVICE_TASK_TYPE.equals(bpmnActivity.getActivityType())) {
             bpmnActivityEntity = entityManager.find(BPMNActivityEntity.class, pkId);
         }
 
         if (bpmnActivityEntity == null) {
-            bpmnActivityEntity = createBpmnActivityEntity(event);
+            return createBpmnActivityEntity(event);
+        } else {
+            return Optional.of(bpmnActivityEntity);
         }
-
-        return bpmnActivityEntity;
     }
 
-    public BaseBPMNActivityEntity createBpmnActivityEntity(CloudRuntimeEvent<?, ?> event) {
+    public Optional<BaseBPMNActivityEntity> createBpmnActivityEntity(CloudRuntimeEvent<?, ?> event) {
```

Comment:
The fact that a method called `createBpmnActivityEntity` returns a nullable `Optional` is a bit weird. If we don't have any better solutions then we should at least add some JavaDoc to this method to explain when it returns an empty Optional (in case of Service Tasks).